### PR TITLE
docs: update go sdk

### PR DESCRIPTION
**Pull request by Anyc66666666**
## What's Changed in this PR

update v0.3,v0.4 docs gosdk

## Checklist

- [x] Please ensure that the content in `summary.yml` matches the current document structure when you changed the document structure.
- [x] This change requires follow-up update in localized docs.

### CODE REVIEW

File: docs/en/v0.4/user-guide/query-data/sdk-libraries/go.md [hunk 1]
```diff
@@ -8,36 +8,38 @@ To begin with, you have to prepare a `QueryRequest`, and you can retrieve data f
...
 
-    resMetric, err := client.Query(context.Background(), queryRequest)
-    if err != nil {
-        fmt.Printf("fail to query, err: %+v\n", err)
-        return
-    }
+	//Query with metric via Sql, and you can also do it via PromQL
+	queryRequest := greptime.QueryRequest{}
+	// if you want to specify another database, you can specify it via: `WithDatabase(database)`
+	queryRequest.WithSql("SELECT * FROM monitor") // .WithDatabase(database)
```
Comment (yuanbohan) [inline docs/en/v0.4/user-guide/query-data/sdk-libraries/go.md:23]: the upper comments can be omitted

File: docs/en/v0.4/user-guide/write-data/sdk-libraries/go.md [hunk 2]
```diff
@@ -23,31 +23,49 @@ then, you can add one `Series` into `Metric`, then create an `InsertRequest` and
...
+	insertsRequest.Append(insertRequest1).Append(insertRequest2)
+
+	// if you want to insert into different table in one request, you can construct
+	// another InsertRequest, and include it via: insertsRequest.Insert(insertRequest)
+
+	// if you want to specify another database, you can specify it via: `WithDatabase(database)`
+	// insertsRequest.WithDatabase("your database")
+
+	// Fire the real Insert request and Get the affected number of rows
+	res, err := client.Insert(context.Background(), insertsRequest)
```
Comment (yuanbohan) [inline docs/en/v0.4/user-guide/write-data/sdk-libraries/go.md:62]: too many comments here, most of the comments can be omitted

**Review by yuanbohan — APPROVED** [review]
LGTM

File: docs/en/v0.3/user-guide/write-data/sdk-libraries/go.md [hunk 3]
```diff
@@ -23,31 +23,31 @@ then, you can add one `Series` into `Metric`, then create an `InsertRequest` and
...
+	series := greptime.Series{}
+	series.AddStringTag("host", "localhost")
+	series.AddFloatField("cpu", 0.90)
+	series.AddIntField("memory", 1024)
+	series.SetTimestamp(time.Now())
+
+	metric := greptime.Metric{}
+	metric.AddSeries(series)
+
+	insertRequest1 := greptime.InsertRequest{}
```
Comment (nicecui) [inline docs/en/v0.3/user-guide/write-data/sdk-libraries/go.md:35]: it isn't a good variable name

File: docs/en/v0.3/user-guide/write-data/sdk-libraries/go.md [hunk 4]
```diff
@@ -23,31 +23,31 @@ then, you can add one `Series` into `Metric`, then create an `InsertRequest` and
...
+	insertRequest1.WithTable("monitor").WithMetric(metric)
+
+	insertRequest2 := greptime.InsertRequest{}
+	insertRequest2.WithTable("temperature").WithMetric(metric)
```
Comment (nicecui) [inline docs/en/v0.3/user-guide/write-data/sdk-libraries/go.md:39]: insert the same metric to tables `monitor` and `temperature`, that example isn't good.

File: docs/zh/v0.3/user-guide/clients/sdk-libraries/go.md [hunk 5]
```diff
@@ -3,7 +3,7 @@
 ## 安装 SDK
 
 ```sh
-go get github.com/GreptimeTeam/greptimedb-client-go
+go get github.com/GreptimeTeam/greptimedb-client-go@v0.1.2
```
Comment (nicecui) [inline docs/zh/v0.3/user-guide/clients/sdk-libraries/go.md:6]: Is SDK v0.1.2 compatible with DB v0.3?

Comment (Anyc66666666) [reply docs/zh/v0.3/user-guide/clients/sdk-libraries/go.md:6]: yes

File: docs/en/v0.3/user-guide/write-data/sdk-libraries/go.md [hunk 6]
```diff
@@ -23,31 +23,32 @@ then, you can add one `Series` into `Metric`, then create an `InsertRequest` and
...
+	series.SetTimestamp(time.Now())
+
+	metric := greptime.Metric{}
+	metric.AddSeries(series)
+
+	monitorReq := greptime.InsertRequest{}
+	monitorReq.WithTable("monitor").WithMetric(metric)
+
+	monitorCopyReq := greptime.InsertRequest{}
+	monitorCopyReq.WithTable("monitor_copy").WithMetric(metric)
```
Comment (yuanbohan) [inline docs/en/v0.3/user-guide/write-data/sdk-libraries/go.md:39]: maybe it's better to omit the multiple table examples in greptime docs. The detail usage can be organized in greptimedb-client-go, users can find more examples there. While in docs, it's better to just tell user the whole procedure, and help user to get a correct way to interact with GreptimeDB.

**Review by yuanbohan — APPROVED** [review]

File: docs/en/v0.3/greptimecloud/integrations/sdk-libraries/go.md [hunk 7]
```diff
@@ -19,8 +19,8 @@ options := []grpc.DialOption{
 
 cfg := greptime.NewCfg("<host>").
     WithDatabase("<dbname>").
-    WithPort(4001).
-    WithAuth("<username>", "<password>").
+    WithPort(4001).              // default port
+    WithAuth("<username>", "*Your GreptimeCloud service password*").
```
Comment (nicecui) [inline docs/en/v0.3/greptimecloud/integrations/sdk-libraries/go.md:23]:
```suggestion
    WithAuth("<username>", "<password>").
```
`<password>` can be recognized at console.

File: docs/zh/v0.4/greptimecloud/integrations/sdk-libraries/go.md [hunk 8]
```diff
@@ -20,7 +20,7 @@ options := []grpc.DialOption{
 
 cfg := greptime.NewCfg("<host>").
     WithDatabase("<dbname>").
-    WithPort(4001).
+    WithPort(4001).              // default port
     WithAuth("<username>", "*Your GreptimeCloud service password*").
```
Comment (nicecui) [inline docs/zh/v0.4/greptimecloud/integrations/sdk-libraries/go.md:24]:
```suggestion
    WithAuth("<username>", "<password>").
```

**Review by nicecui — APPROVED** [review]

**Pull request merged**